assets/src/edit-story/app/story/useStoryReducer/reducers/addElements.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Add elements to current page.
+ *
+ * Elements are expected to a be list of element objects with at least an id property.
+ * No validation is made on these objects. If elements aren't a list, nothing happens.
+ *
+ * Elements will be added to the front (end) of the list of elements on the current page.
+ *
+ * Selection is set to be exactly the new elements.
+ *
+ * @param {Object} state Current state
+ * @param {Object} payload Action payload
+ * @param {Array.<Object>} payload.elements Elements to insert on the given page.
+ * @return {Object} New state
+ */
+function addElements( state, { elements } ) {
+	const pageIndex = state.pages.find( ( { id } ) => id === state.current );
+	const oldPage = state.pages[ pageIndex ];
+	const newPage = {
+		...oldPage,
+		elements: [
+			...oldPage.elements,
+			...elements,
```

Comment:
Would elements always be added to the end, e.g. in case of duplication as well? Perhaps that's a UX question, just wondering if there should be an optional index. Probably not important until we actually need it.

Reply:
I considered it but didn't see a clear use-case for it. Maybe if the user cut-pastes in the layer panel, but we might want to just use `arrange` for that.